lib/ClangImporter/ClangImporter.cpp

```diff
@@ -6398,6 +6420,14 @@ CxxRecordSemantics::evaluate(Evaluator &evaluator,
     return CxxRecordSemanticsKind::Iterator;
   }
 
+  if (hasCopyTypeOperations(cxxDecl)) {
+    return CxxRecordSemanticsKind::Owned;
+  }
+
+  if (hasMoveTypeOperations(cxxDecl)) {
+    return CxxRecordSemanticsKind::MoveOnly;
+  }
+
   if (hasPointerInSubobjects(cxxDecl)) {
```

Comment:
So, right now an implicit copy constructor will trigger one of the above cases, which means this case won't be hit for anything other than pointers. This should be hit for cases like `StringRef` where there's no explicit copy ctor.